examples/notebooks/beam-ml/run_inference_tensorflow_with_tensorflowhub.ipynb

```diff
@@ -0,0 +1,620 @@
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "from apache_beam.ml.inference.base import KeyedModelHandler\n",
+        "from google.protobuf import text_format\n",
+        "import tensorflow as tf\n",
+        "\n",
+        "class FormatOutputKeyed(FormatOutput):\n",
+        "  # To simplify, inherit from FormatOutput.\n",
+        "  def process(self, tuple_in: Tuple):\n",
+        "    key, element = tuple_in\n",
+        "    output = super().process(element)\n",
+        "    yield \"{} : {}\".format(key, output)\n",
+        "\n",
+        "examples = numpy.array([(1,20), (2,40), (3,60), (4,90)], dtype=numpy.float32)\n",
+        "keyed_model_handler = KeyedModelHandler(TFModelHandlerNumpy(save_model_dir_multiply))\n",
+        "with beam.Pipeline() as p:\n",
+        "    _ = (p | 'CreateExamples' >> beam.Create(examples)\n",
+        "           | RunInference(keyed_model_handler)\n",
+        "           | beam.ParDo(FormatOutputKeyed())\n",
+        "           | beam.Map(print)\n",
+        "        )"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "KPtE3fmdJQry",
+        "outputId": "e1a4cb24-458c-4cd1-8f58-5e72351a71f6"
+      },
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "1.0 : example is 20.0 prediction is [24.687366]\n",
+            "2.0 : example is 40.0 prediction is [47.377365]\n",
+            "3.0 : example is 60.0 prediction is [70.06737]\n",
+            "4.0 : example is 90.0 prediction is [104.10237]\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## RunInference with Tensorflow Hub\n",
+        "\n",
+        "To use tensorflow hub's trained model URL, pass it to the `model_uri` field of TFModelHandler class."
+      ],
+      "metadata": {
+        "id": "dVLTtFxDuJT_"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import tensorflow_hub as hub"
+      ],
+      "metadata": {
+        "id": "H4-ZvkcTv7MO"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# URL of the trained model from tensorflow hub\n",
+        "CLASSIFIER_URL =\"https://tfhub.dev/google/tf2-preview/mobilenet_v2/classification/4\""
+      ],
+      "metadata": {
+        "id": "n3M6FNaUwBbl"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import numpy as np\n",
+        "import PIL.Image as Image\n",
+        "\n",
+        "IMAGE_RES = 224\n",
+        "img = tf.keras.utils.get_file(origin='https://images-na.ssl-images-amazon.com/images/G/01/img15/pet-products/small-tiles/23695_pets_vertical_store_dogs_small_tile_8._CB312176604_.jpg')\n",
```

Comment:
Do you know how this image is licensed? Generally, we try to use images that are available under creative commons licensing since that makes licensing easier, we should also call that out here. See https://github.com/apache/beam/blob/master/examples/notebooks/beam-ml/run_inference_multi_model.ipynb for example

Reply:
Changed to use the same image used in this notebook.